src/fieri/app/models/metrics_runner.rb

```diff
@@ -24,4 +24,8 @@ def cookbook_api_response(params)
   def cookbook_version_api_response(params)
     SupermarketApiRunner.new.cookbook_version_api_response(params['cookbook_name'], params['cookbook_version'])
   end
+
+  def source_repo(cookbook_json)
+    cookbook_json.match(%r{(?<=github.com\/)[\w-]+\/[\w-]+}).to_s
```

Comment:
I ended up going with SourceRepoMetric - I think it's the best we can do for now.

As for your second, I can access the full github url with 
```
JSON.parse(cookbook_json)['source_url']
```
which will return something like "http://github.com/some_user/some_repo" , but I still will need to parse it with a regex because the github url will only accept "some_user/some_repo" when using the contents api.